src/globus_cli/commands/timer/_common.py

```diff
@@ -0,0 +1,88 @@
+import datetime
+import re
+from typing import Any, Dict, Optional, cast
+from urllib.parse import urlparse
+
+import click
+
+# List of datetime formats accepted as input. (`%z` means timezone.)
+DATETIME_FORMATS = [
+    "%Y-%m-%d",
+    "%Y-%m-%d %H:%M:%S",
+    "%Y-%m-%d %H:%M:%S%z",
+    "%Y-%m-%dT%H:%M:%S",
+    "%Y-%m-%dT%H:%M:%S%z",
+    "%Y-%m-%dT%H:%M:%S.%f%z",
+]
```

Comment:
Timer will only return the last one, but the default set of click formats is actually not a lot—for instance, none of the defaults allow including a timezone or offset (which is required, because the server does).

Reply:
> Timer will only return the last one

If we know this, I don't understand why we don't just do `strptime` for that?

If we're optimizing not for LOC but for simplicity, isn't that the shortest possible solution?

Another option would be something like this script (which is complete so that I could test it trivially).
```python
import datetime
import sys


def parsetime(s: str) -> datetime.datetime:
    try:
        return datetime.datetime.strptime(s, "%Y-%m-%dT%H:%M:%S.%f%z")
    except ValueError:
        return datetime.datetime.strptime(s, "%Y-%m-%dT%H:%M:%S%z")


if __name__ == "__main__":
    assert len(sys.argv) == 2
    parsetime(sys.argv[1])
    print("ok")
```

Reply:
These are meant to be input, not output formats. However I want to assume that a user can pass exactly a time they received from the API back into the CLI and it will work.

Reply:
> I want to assume that a user can pass exactly a time they received from the API back into the CLI and it will work.

That's a very sensible goal, and it makes sense of this.

I'm still a little bit bothered by using `click.DateTime` inside of an application for parsing an API response. e.g. Consider what happens if the API unexpectedly sends back `"None"` and parsing fails.

What about this as a resolution?
- move `DATE_FORMATS` to `globus_cli.constants`
- define a helper, either in this module or in `globus_cli.utils`, which does the iterative try-except `datetime.strptime` calls, and raises a `ValueError` on bad strings
- plan (not for today) to add a date-time type to `globus_cli.parsing` which provides `click.DateTime(formats=constants.DATE_FORMATS)` and use that *everywhere* in the CLI

Reply:
This sounds good—I just added this version here: https://github.com/globus/globus-cli/pull/651/files#diff-5f525522744cd4f374591971e59bfedcffa89185db900559a9d211f5f7939d62R58

Are you good with this, moved to `globus_cli.utils`/`globus_cli.constants`?